src/providers/ec2/ec2.go

```diff
@@ -101,3 +145,45 @@ func (p provider) ShouldRetry() bool {
 func (p *provider) BackoffDuration() time.Duration {
 	return util.ExpBackoff(&p.backoff, maxBackoff)
 }
+
+func fetchBody(client *http.Client, url string, acceptedStatuses ...int) ([]byte, int, error) {
```

Comment:
I originally wrote the function so that the consumer just had to worry about the body and the error, but I quickly found that the consumers need to make decisions based on the status code.

I agree with the logging.